### DISCUSSION TITLE
E2: Remove duplicate directive error for includes?

### DISCUSSION THREAD

**Discussion by shadowscion** [wiremod/wire#2839] [Suggestions]
Talking about these errors here:

```Directive (@persist) contains multiple definitions of the same variable```

It would be great if the limitation could be removed when including a file, barring any technical reasons for having it there. As it is now, cross-including and libraries that need to span multiple files (that have to use the same variables) are not easy to write or convenient to use. 

**Edit:**
I'm not always the best with words, so I'm going to post an example from the discord by Divran as to why I think this should be considered:

> if multiple includes add different feature sets of the same main functionality, but having all of them is not required. you can pick and choose which includes you want in your project because all of them will define the variables

and also

> the multiple def check does literally nothing. its only there to help you fix mistakes in your code
> because at the time it was made, includes didnt exist
> but now that includes exists we can change it to only say this if you have multiple defs of the same variable name but different types, so if one file has @persist A:vector and another file has @persist A:angle then that's a conflict and should be fixed

Currently there are ways around this limitation (such as defining the variables ahead of the main code in each file) but they are not very pretty.

**Comment by AbigailBuccaneer** [comment]
The checks exist for `@persist`, `@inputs`, `@outputs` and `@trigger`. Should redefinitions in all of them be allowed?

**Comment by shadowscion** [comment]
Just to be consistent I would say yes, I can't think of any reasons not to. From what I could tell and what I was told in the discord, these checks don't actually do anything other than force formatting rules on your code.

**Comment by shadowscion** [comment]
Has any further consideration been put into this? Dirty workarounds just bloat code into unreadable territory.

**Comment by AbigailBuccaneer** [comment]
I'm happy for the checks to be removed. Part of me thinks that duplicate `@inputs`/`@outputs`/`@trigger` should be warnings (and obviously errors if the types are inconsistent). I just don't think anybody's gotten around to doing the work and making a pull request yet.

**Comment by thegrb93** [comment]
Pretty sure it's to avoid bugs that would pop up with scripts using same var names.

**Comment by shadowscion** [comment]
Can this error use the new warning system instead?

**Comment by Denneisk** [comment]
Implemented in #2869

**Comment by shadowscion** [comment]
At last
